app/controllers/spree/api/v2/storefront/wished_products_controller.rb

```diff
@@ -0,0 +1,74 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class WishedProductsController < Spree::Api::V2::BaseController
+
+          helper Spree::Wishlists::ApiHelpers
+
+          def create
+            authorize! :create, Spree::WishedProduct
+
+            @wished_product = Spree::WishedProduct.new(wished_product_attributes)
+
+            current_wishlist_user = if params[:user_id] && @current_user_roles.include?('admin')
```

Comment:
this would be security risk, we need to implement ability..
we can only show public wishlists to anyone..by user id..
private wishlist would not be shared